dev/src/write-batch.ts

```diff
@@ -603,25 +595,26 @@ export class WriteBatch implements firestore.WriteBatch {
    * this request.
    * @returns  A Promise that resolves when this batch completes.
    */
-  async _commit(commitOptions?: {
+  async _commit<Req, Resp>(commitOptions?: {
     transactionId?: Uint8Array;
     requestTag?: string;
-  }): Promise<WriteResult[]> {
+    retryCodes?: number[];
+    methodName?: FirestoreUnaryMethod;
+  }): Promise<Resp> {
     // Note: We don't call `verifyNotCommitted()` to allow for retries.
-    this._committed = true;
+    this.state = BatchState.SENT;
 
     const tag = (commitOptions && commitOptions.requestTag) || requestTag();
     await this._firestore.initializeIfNeeded(tag);
 
-    const database = this._firestore.formattedName;
-    const explicitTransaction = commitOptions && commitOptions.transactionId;
-
-    const request: api.ICommitRequest = {
-      database,
+    const request = {
+      database: this._firestore.formattedName,
       writes: this._ops.map(op => op.op()),
+      transaction: commitOptions?.transactionId,
     };
-    if (commitOptions?.transactionId) {
-      request.transaction = commitOptions.transactionId;
+
+    if (request.transaction === undefined) {
+      delete request.transaction;
```

Comment:
The old way seemed better. Just add the transaction if it is present. To make this typesafe, you can do something like this: https://paste.googleplex.com/5733759430164480

Reply:
done. Added comment and removed cast to `unknown` below.